pkg/initializers/dataset/cache_initalizer.py

```diff
@@ -0,0 +1,209 @@
+            logging.error("CACHE_IMAGE environment variable is required but not provided")
+            raise ValueError("CACHE_IMAGE environment variable is required")
+        
+        if not self.config.iam_role:
+            logging.error("IAM_ROLE environment variable is required but not provided")
+            raise ValueError("IAM_ROLE environment variable is required")
+
+        train_job_name = self.config.train_job_name or "cache-test"
+        cache_image = self.config.cache_image
+        cluster_size = self.config.cluster_size or "3"
+        iam_role = self.config.iam_role
+        head_cpu = self.config.head_cpu or "1"
+        head_mem = self.config.head_mem or "1Gi"
+        worker_cpu = self.config.worker_cpu or "2"
+        worker_mem = self.config.worker_mem or "2Gi"
+        
+        substitutions = {
+            'NAME': train_job_name,
+            'IAM_ROLE': iam_role,
+            'SIZE': cluster_size,
+            'IMAGE': cache_image,
+            'HEAD_CPU': head_cpu,
+            'HEAD_MEM': head_mem,
+            'WORKER_CPU': worker_cpu,
+            'WORKER_MEM': worker_mem,
+        }
+        
+        if self.config.metadata_loc:
+            substitutions['METADATA_LOC'] = self.config.metadata_loc
+        if self.config.table_name:
+            substitutions['TABLE_NAME'] = self.config.table_name
+        if self.config.schema_name:
+            substitutions['SCHEMA_NAME'] = self.config.schema_name
+
+        deploy_lws_with_substitution(
+            train_job_name,
+            'cache-initializer-template.yaml',
+            namespace='cache-test',
+            substitutions=substitutions,
+            timeout=600
+        )
+
+        logging.info("Cache dataset initialization completed")
+
+
+def deploy_lws_with_substitution(train_job_name, yaml_path, config_file: Optional[str] = None, namespace='default', substitutions=None,
+                                 timeout=300):
+    """
+    Deploys a parameterized LeaderWorkerSet YAML with ServiceAccount, environment substitution,
+    and waits for deployment readiness.
+
+    Args:
+        train_job_name(str): train_job_name
+        yaml_path (str): Path to YAML file containing configuration
+        namespace (str): Target Kubernetes namespace
+        substitutions (dict): Additional variables for substitution
+        timeout (int): Maximum wait time in seconds
+
+    Returns:
+        bool: True if deployment succeeded and became ready
+    """
+    with open(yaml_path, 'r') as f:
+        content = f.read()
+
+    sub_vars = {**os.environ, **(substitutions or {})}
+
+    for key, value in sub_vars.items():
+        content = content.replace(f'${key}', value)
+        content = content.replace(f'${{{key}}}', value)
+
+    resources = list(yaml.safe_load_all(content))
+
+    if config_file or not is_running_in_k8s():
+        config.load_kube_config(config_file=config_file)
+    else:
+        config.load_incluster_config()
```

Comment:
This is not needed since initializer always runs in Kubernetes (e.g. `config.load_incluster_config()`)